### DISCUSSION TITLE
[v4] @theme --background-size-name not working.

### DISCUSSION THREAD

**Discussion by demonguyj** [tailwindlabs/tailwindcss#15386] [Help]
**What version of Tailwind CSS are you using?**
4.0.0 beta.6

**What build tool (or framework if it abstracts the build tool) are you using?**

Next.js 15.1.0

**What version of Node.js are you using?**

For example: v20.18.0

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

A Tailwind Play link or public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways.

A reproduction is **required** when filing an issue — any issue opened without a reproduction will be closed and you'll be asked to create a new issue that includes a reproduction. We're a small team and we can't keep up with the volume of issues we receive if we need to reproduce each issue from scratch ourselves.

**Describe your issue**

I wrote 
@theme {
    --background-size-full: 100%;
}

It is expected to:

.bg-full {
   background-size:100%;
}

but it is not created.

**Comment by wongjn** [comment]
The background size utilities are registered as static utilities:

https://github.com/tailwindlabs/tailwindcss/blob/42778577021e44f07233494366a3b49e9f5eab41/packages/tailwindcss/src/utilities.ts#L2271-L2273

Which means you can't configure new ones in `@theme`. However, you could add your own `bg-full` via a CSS rule:

```css
.bg-full {
  background-size:100%;
}
```
Or as a utility:

```css
@utility bg-full {
  background-size:100%;
}
```

**Reply by demonguyj** [reply]
I created 

@utility bg-full {
  background-size:100%;
}

And I tried apply to before:bg-full.
It isn't worked...

But .bg-full is working.
